src/borg/repository.py

```diff
@@ -1084,6 +1093,9 @@ class LoggedIO:
     class SegmentFull(Exception):
         """raised when a segment is full, before opening next"""
 
+    class AtticSegment(Error):
+        """Attic segment detected [segment {}]"""
```

Comment:
The Attic MAGIC will always be at offset 0, so I think it is fine to hardcode the value here.